build/xtask/src/dist.rs

```diff
@@ -130,80 +138,66 @@ pub fn package(
         &remap_paths,
     )?;
 
-    // Build all relevant tasks, collecting entry points into a HashMap
-    let entry_points: HashMap<String, u32> = toml
+    // Build all relevant tasks, collecting entry points into a HashMap.  If
+    // we're doing a partial build, then assign a dummy entry point into
+    // the HashMap, because the kernel kconfig will still need it.
+    let entry_points: HashMap<_, _> = toml
         .tasks
         .keys()
-        .filter(|name| {
-            tasks_to_build
-                .as_ref()
-                .map(|m| m.contains(name))
-                .unwrap_or(true)
-        })
         .map(|name| {
-            build_task(
-                &toml,
-                name,
-                verbose,
-                edges,
-                &allocs,
-                &dist_dir,
-                &remap_paths,
-                &mut all_output_sections,
-            )
-            .map(|ep| (name.clone(), ep))
+            let ep = if tasks_to_build.contains(name) {
+                build_task(
+                    &toml,
+                    name,
+                    verbose,
+                    edges,
+                    &allocs,
+                    &dist_dir,
+                    &remap_paths,
+                    &mut all_output_sections,
+                )
+            } else {
+                Ok(allocs.tasks[name]["flash"].start)
+            };
+            ep.map(|ep| (name.clone(), ep))
         })
         .collect::<Result<_, _>>()?;
 
-    // If we've done a partial build, we can't do the rest because we're missing
-    // required information, so, escape.
+    // Build the kernel!
+    let kern_build = if tasks_to_build.contains("kernel") {
+        Some(build_kernel(
+            &toml,
+            verbose,
+            edges,
+            &allocs,
+            &mut all_output_sections,
+            &entry_points,
+            &dist_dir,
+            &remap_paths,
+        )?)
+    } else {
+        None
+    };
+
+    // If we've done a partial build (which may have included the kernel), bail
+    // out here before linking stuff.
     if partial_build {
         return Ok(());
     }
 
-    // Build the kernel!
-    let (kentry, ksymbol_table) = build_kernel(
-        &toml,
-        verbose,
-        edges,
-        &allocs,
-        &mut all_output_sections,
-        &entry_points,
-        &dist_dir,
-        &remap_paths,
-    )?;
-
     // Generate combined SREC, which is our source of truth for combined images.
+    let (kentry, ksymbol_table) = kern_build.unwrap();
```

Comment:
So, i asked myself, "why this unwrap?" Answer: this function returns `Result`, and this is an Option. Makes sense.

But if we look at things above... `build_kernel` returns a `Result`, which we `?` and then wrap in an option, only to unwrap it here. So I was like "Could we get rid of that extra layer of indirection?" And after trying some things, the answer is "no". Or at least, I couldn't figure out something better. Kudos on finding a legit situation for unwrap 😅

Reply:
Alright, I figured out a way to do it without repeating anything or invoking `unwrap()`, but I think it's worse-looking:
```rust
    // Build the kernel!
    let kern_build = if tasks_to_build.contains("kernel") {
        Some(build_kernel(
            &cfg,
            &allocs,
            &mut all_output_sections,
            &entry_points,
        )?)
        .filter(|_| !partial_build) // <-- this is new
    } else {
        None
    };

    // If we've done a partial build (which may have included the kernel), bail
    // out here before linking stuff.
    let kern_build = match kern_build { // <-- this is new
        Some(k) => k,
        None => return Ok(()),
    };

    // Generate combined SREC, which is our source of truth for combined images.
    let (kentry, ksymbol_table) = kern_build;
```